core/ledger/kvledger/pvtrwstorage/transient_store.go

```diff
@@ -16,26 +16,154 @@ limitations under the License.
+// interface.
+type transientStoreProvider struct {
+	dbProvider *leveldbhelper.Provider
+}
+
+// TransientStoreProvider provides an instance of a TransientStore
 type TransientStoreProvider interface {
-	OpenStore(namespace string) (TransientStore, error)
+	OpenStore(dbName string) (TransientStore, error)
 	Close()
 }
 
-// EndorserTxRWSets contains the private rwset along with the id of the producing endorser
-type EndorserTxRWSets struct {
-	endorserid string
-	rwset      []byte
+// TransientStore holds an instance of a levelDB.
+type transientStore struct {
+	db     *leveldbhelper.DBHandle
+	dbName string
+}
+
+type rwsetScanner struct {
+	txId  string
+	dbItr iterator.Iterator
 }
 
 // TransientStore manages the storage of private read-write sets for a namespace for sometime.
 // Ideally, a ledger can remove the data from this storage when it is committed to the permanent storage or
 // the pruning of some data items is enforced by the policy
 type TransientStore interface {
-	Persist(endorserid string, txRWSet *TxRWSet, endorsementBlkHt uint64) error
-	GetTxRWSetByTxid() []*EndorserTxRWSets
+	Persist(txid string, endorserid string, endorsementBlkHt uint64, privateSimulationResults []byte) error
+	GetTxPrivateRWSetByTxId(txId string) (commonledger.ResultsIterator, error)
 	Purge(maxBlockNumToRetain uint64) error
 	GetMinEndorsementBlkHt() (uint64, error)
 	Shutdown()
 }
+
+// NewTransientStoreProvider instantiates TransientStoreProvider
+func NewTransientStoreProvider() TransientStoreProvider {
+	dbPath := ledgerconfig.GetTransientStorePath()
+	dbProvider := leveldbhelper.NewProvider(&leveldbhelper.Conf{DBPath: dbPath})
+	return &transientStoreProvider{dbProvider: dbProvider}
+}
+
+// OpenStore returns a handle to a name database in TransientStore
+func (provider *transientStoreProvider) OpenStore(dbName string) (TransientStore, error) {
+	dbHandle := provider.dbProvider.GetDBHandle(dbName)
+	return &transientStore{db: dbHandle, dbName: dbName}, nil
+}
+
+// Close closes the TransientStoreProvider
+func (provider *transientStoreProvider) Close() {
+	provider.dbProvider.Close()
+}
+
+// Persist stores the private read-write set of a transaction in the transientStore
+func (store *transientStore) Persist(txId string, endorserId string,
+	endorsementBlkHt uint64, privateSimulationResults []byte) error {
+
+	// Create compositeKey with the namespace of private read-write set, txId and endorserId
+	compositeKey := CreateCompositeKeyForPRWSet(PRWSET_NS, txId, endorserId, endorsementBlkHt)
+	if err := store.db.Put(compositeKey, privateSimulationResults, true); err != nil {
+		return err
+	}
+
+	// Create compositeKey with endorsementBlkHt, txId, endorserId & Store the compositeKey (purge index)
+	// in a separate namespace with a null byte as value.
+	compositeKey = CreateCompositeKeyForPurgeIndex(PURGEINDEX_NS, endorsementBlkHt, txId, endorserId)
+	if err := store.db.Put(compositeKey, emptyValue, true); err != nil {
+		return err
+	}
+	return nil
```

Comment:
Better to use a batch instead of separately calling the put two times (efficiency and consistency)